src/main/kotlin/view/FilterPanel.kt

```diff
@@ -128,27 +134,47 @@ class FilterPanel : View() {
                                             addClass(CssStyle.labelTag)
                                         }
                                         val slider = slider {
-                                            min = 0.0
+                                            min = -100.0
                                             max = 100.0
+                                            isShowTickMarks = true
+                                            majorTickUnit = 50.0
+                                            minorTickCount = 1
+                                            blockIncrement = 1.0
                                         }
                                         sliders += slider
-                                        slider.value = 50.0
-                                        slider.valueChangingProperty()
-                                            .addListener(ChangeListener { _, _, _ -> op(slider.value / 50) })
-
+                                        slider.value = 0.0
+                                        slider.valueProperty()
+                                            .addListener(ChangeListener { _, _, _ -> op(slider.value / 100 + 1) })
                                         addClass(CssStyle.filterSlider)
+
+                                        val spinner = spinner(
+                                            -100.0,
+                                            100.0,
+                                            0.0,
+                                            1.0,
+                                            true,
+                                            doubleProperty(0.0)
+                                        ) {
+                                            maxWidth = 70.0
+                                        }
+
+//                                        spinner.valueFactory.converter =
+//                                            (NumberStringConverter("#.0"))
```

Comment:
The spinner now kind of works, still remain some issues:

- [ ] If user delete the value in the spinner, will throw NPE.
- [ ] Still finding a way to convert into no decimal points, however since slider is in Double, it can have type convert issue using this line of code..